File: src/p0019/java/solution.java
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode sentinel = new ListNode(0);
        sentinel.next = head;
        ListNode first = sentinel;
        for (int i = 0; i < n; i++) {
            first = first.next;
        }
        ListNode second = sentinel;
        while (first.next != null) {
            first = first.next;
            second = second.next;
        }
        ListNode removeNode = second.next;
        second.next = removeNode.next;
        return sentinel.next;
    }
}

File: src/p0019/python/solution.py
class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        sentinel = ListNode(0)
        sentinel.next = head
        first = sentinel
        for _ in range(n):
            first = first.next
        second = sentinel
        while first.next:
            first = first.next
            second = second.next
        remove_node = second.next
        second.next = remove_node.next
        return sentinel.next
